packages/functional-tests/tests/react-conversion/signup.spec.ts

```diff
@@ -46,6 +44,8 @@ test.describe('severity-1 #smoke', () => {
     }) => {
       await signupReact.goto();
       await signupReact.fillOutEmailFirst(email);
+      await page.waitForURL(/signup/);
+      expect(await page.locator('#root').isEnabled()).toBeTruthy();
```

Comment:
I don't think I understand what 'enabled' means for the root element.  Is it always enabled?

Reply:
Thanks for the callout. Just checked through our codebase and there's some inconsistency in how we check for `#root` (only present in react pages vs backbone that has `#stage`) and what I used here is definitely the clunkier option. In this kind of context, `locator` does not wait for state (visible, enabled) unless an action is performed on the locator, so checking for `isEnabled` or `isVisible` is necessary.

On a React page,
`expect(await page.locator('#stage')).toBeTruthy();` passes even though there is no such element. (false positive)

`expect(await page.locator('#stage').isEnabled()).toBeTruthy();` fails after a long timeout.

`waitForSelector` that is used elsewhere in our tests might be a better fit even though [it's discouraged](https://playwright.dev/docs/api/class-page#page-wait-for-selector), because the preferred web assertions that assert visibility don't work for ids.